src/main/java/Duke.java

```diff
@@ -1,10 +1,126 @@
+import java.util.Scanner;
+
 public class Duke {
+    //Application Logo
+    private static final String LOGO = " ____        _        \n"
+            + "|  _ \\ _   _| | _____ \n"
+            + "| | | | | | | |/ / _ \\\n"
+            + "| |_| | |_| |   <  __/\n"
+            + "|____/ \\__,_|_|\\_\\___|\n";
+
+    //Strings to define command type
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST_TASKS = "list";
+    private static final String COMMAND_MARK_TASK_AS_DONE = "done";
+    private static final String COMMAND_ADD_TODO_TASK = "todo";
+    private static final String COMMAND_ADD_DEADLINE_TASK = "deadline";
+    private static final String COMMAND_ADD_EVENT_TASK = "event";
+
+    //Output Messages
+    private static final String MESSAGE_START_APPLICATION = "Hello from\n" + LOGO + System.lineSeparator() + "What can I do for you?";
+    private static final String MESSAGE_EXIT_APPLICATION = "Thank you for using our application. We hope to see you again soon";
+
+    //Default values for tasks
+    private static final String DEFAULT_DEADLINE_TIME_CONTENT = "No deadline provided";
+    private static final String DEFAULT_EVENT_TIME_CONTENT = "No date provided";
+
+    private static TaskList task = new TaskList();
+
+    private static void processInput(String input) {
+        final String[] commandTypeAndParams = splitCommandWordAndArgs(input);
+        final String commandType = commandTypeAndParams[0];
+        final String commandArgs = commandTypeAndParams[1];
+        switch (commandType) {
+        case COMMAND_EXIT:
+            exitProgram();
+            break;
+        case COMMAND_LIST_TASKS:
+            listAllTasks();
+            break;
+        case COMMAND_MARK_TASK_AS_DONE:
+            markTaskAsDone(commandArgs);
+            break;
+        case COMMAND_ADD_TODO_TASK:
+            addTodoTaskToList(commandArgs);
+            break;
+        case COMMAND_ADD_DEADLINE_TASK:
+            addDeadlineTaskToList(commandArgs);
+            break;
+        case COMMAND_ADD_EVENT_TASK:
+            addEventTaskToList(commandArgs);
+            break;
+        default:
+            addTaskToList(input);
+        }
+    }
+
+    private static String[] splitCommandWordAndArgs(String rawUserInput) {
+        final String[] split = rawUserInput.trim().split(" ", 2);
+        return split.length == 2 ? split : new String[]{split[0], ""}; // else case: no parameters
+    }
+
+    private static void exitProgram() {
+        PrintUtils.printHorizontalLine(100);
```

Comment:
horizontal line seems to have fixed size. Can consider just setting the line to default at length 100